github-pullrequest-plugin/src/main/java/org/jenkinsci/plugins/github/pullrequest/trigger/check/PullRequestToCauseConverter.java

```diff
@@ -44,10 +41,15 @@ public static PullRequestToCauseConverter toGitHubPRCause(GitHubPRRepository loc
 
     @Override
     public GitHubPRCause apply(final GHPullRequest remotePR) {
-        return from(trigger.getEvents())
-                .transform(toCause(remotePR))
-                .filter(notNull())
-                .firstMatch(new SkippedCauseFilter(listener)).orNull();
+        for (GitHubPREvent event: trigger.getEvents()) {
```

Comment:
I think the best variant would be migrate from guava to java 8. 
If logic would be in Trigger, then it would be more difficult to test the code. So having granular logic sounds better. Then doing skip filtering in Converter looks good. And then i think it would make sense to rename class (or just better document it).